sessionctx/varsutil/varsutil.go

```diff
@@ -39,16 +56,25 @@ func GetSessionSystemVar(s *variable.SessionVars, key string) (string, error) {
 	case variable.TiDBCurrentTS:
 		return fmt.Sprintf("%d", s.TxnCtx.StartTS), nil
 	}
-
+	isSQLModeVar := key == variable.SQLModeVar
 	sVal, ok := s.Systems[key]
+	if isSQLModeVar {
+		sVal = formatSQLModeStr(sVal)
```

Comment:
Can we format the SQLModeStr before Set?

Reply:
OK. It's clearer to format SQLModeStr before set, and there's no need to format every time fetch it.